_studio/shared/src/cm_mem_copy.cpp

```diff
@@ -3164,6 +3165,59 @@ mfxStatus CmCopyWrapper::CopyMirrorVideoToVideoMemory(void *pDst, void *pSrc, mf
 
 #define CM_RGB_SUPPORTED_COPY_SIZE(ROI) (ROI.width <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT && ROI.height <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT )
 
+bool CmCopyWrapper::CheckSurfaceContinuouslyAllocated(mfxFrameSurface1 *surf)
+{
+    mfxFrameData* ptrFrameData = &surf->Data;
+
+    switch (surf->Info.FourCC)
+    {
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P016:
+    case MFX_FOURCC_P210:
+    case MFX_FOURCC_NV16:
```

Comment:
P210 and NV16 are 422 formats, unlike 420 their UV size is equal to Y. Don't divide height /2.